Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -1177,13 +1183,17 @@ public void When_no_exception_should_be_thrown_for_async_func_after_wait_time_bu
             //-----------------------------------------------------------------------------------------------------------
             // Arrange
             //-----------------------------------------------------------------------------------------------------------
-            var watch = Stopwatch.StartNew();
             var waitTime = 2.Seconds();
             var pollInterval = 10.Milliseconds();
 
+            var clock = new FakeClock();
+            var timer = clock.StartTimer();
+            clock.Delay(waitTime);
+            clock.CompletesBeforeTimeout();
```

Comment:
This test seems to be the only one requiring this call (note that the RunsIntoTimeout method) also works here.

Reply:
Whether you use `CompletesBeforeTimeout` or `RunsIntoTimeout` seems to only matter for sync assertions, e.g. `CompleteWithin()`.
See `When_task_completes_fast_it_should_succeed`.

I frankly don't known if that is intended.